### PR TITLE
Add KEF services

### PR DESCRIPTION
**Pull request by basnijholt**
## Proposed change

Adds the services that are added in https://github.com/home-assistant/core/pull/31967 to the documentation.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/31967

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/kef.markdown [hunk 1]
```diff
@@ -87,7 +87,89 @@ media_player:
    volume_step: 0.05
 ```
 
-Notes:
+## Advanced - configure DSP settings and services
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
## Services
```

File: source/_integrations/kef.markdown [hunk 2]
```diff
@@ -87,7 +87,89 @@ media_player:
...
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| entity_id              | No       | The entity_id of the KEF speaker. |
+| desk_mode              | Yes      | "Desk mode" (true or false) |
+| wall_mode              | Yes      | "Wall mode" (true or false) |
+| phase_correction       | Yes      | "Phase correction" (true or false) |
+| high_pass              | Yes      | "High-pass mode" (true or false) |
+| sub_polarity           | Yes      | "Sub polarity" ("-" or "+") |
+| bass_extension         | Yes      | "Bass extension" selector ("Less", "Standard", or "Extra") |
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
| Service data attribute | Optional | Description |
| ---------------------- | -------- | ----------- |
| entity_id              | No       | The entity_id of the KEF speaker. |
| desk_mode              | Yes      | "Desk mode" (`true` or `false`) |
| wall_mode              | Yes      | "Wall mode" (`true` or `false`) |
| phase_correction       | Yes      | "Phase correction" (`true` or `false`) |
| high_pass              | Yes      | "High-pass mode" (`true` or `false`) |
| sub_polarity           | Yes      | "Sub polarity" (`-` or `+`) |
| bass_extension         | Yes      | "Bass extension" selector (`Less`, `Standard`, or `Extra`) |
```

File: source/_integrations/kef.markdown [hunk 3]
```diff
@@ -87,7 +87,89 @@ media_player:
...
 ```
 
-Notes:
+## Advanced - configure DSP settings and services
+
+Just like in the KEF Control app, we can change the digital signal processing (DSP) settings.
+
+The speaker's current settings become available as attributes when calling `kef.update_dsp`.
+
+### Service `update_dsp`
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
### Service `kef.update_dsp`
```

File: source/_integrations/kef.markdown [hunk 4]
```diff
@@ -87,7 +87,89 @@ media_player:
...
+
+Update all DSP settings.
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| entity_id              | No       | The entity_id of the KEF speaker. |
+
+### Service `set_mode`
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
### Service `kef.set_mode`
```

File: source/_integrations/kef.markdown [hunk 5]
```diff
@@ -87,7 +87,89 @@ media_player:
...
+| ---------------------- | -------- | ----------- |
+| entity_id              | No       | The entity_id of the KEF speaker. |
+| desk_mode              | Yes      | "Desk mode" (true or false) |
+| wall_mode              | Yes      | "Wall mode" (true or false) |
+| phase_correction       | Yes      | "Phase correction" (true or false) |
+| high_pass              | Yes      | "High-pass mode" (true or false) |
+| sub_polarity           | Yes      | "Sub polarity" ("-" or "+") |
+| bass_extension         | Yes      | "Bass extension" selector ("Less", "Standard", or "Extra") |
+
+### Service `set_desk_db`
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
### Service `kef.set_desk_db`
```

File: source/_integrations/kef.markdown [hunk 6]
```diff
@@ -87,7 +87,89 @@ media_player:
...
+
+Set the "Desk mode" slider of the speaker in dB.
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| entity_id              | No       | The entity_id of the KEF speaker. |
+| db                     | No       | Value of the slider (-6 to 0 with steps of 0.5) |
+
+### Service `set_wall_db`
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
### Service `kef.set_wall_db`
```

File: source/_integrations/kef.markdown [hunk 7]
```diff
@@ -87,7 +87,89 @@ media_player:
...
+
+Set the "Wall mode" slider of the speaker in dB.
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| entity_id              | No       | The entity_id of the KEF speaker. |
+| db                     | No       | Value of the slider (-6 to 0 with steps of 0.5) |
+
+### Service `set_treble_db`
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
### Service `kef.set_treble_db`
```

File: source/_integrations/kef.markdown [hunk 8]
```diff
@@ -87,7 +87,89 @@ media_player:
...
+
+Set desk the "Treble trim" slider of the speaker in dB.
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| entity_id              | No       | The entity_id of the KEF speaker. |
+| db                     | No       | Value of the slider (-2 to 2 with steps of 0.5) |
+
+### Service `set_high_hz`
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
### Service `kef.set_high_hz`
```

File: source/_integrations/kef.markdown [hunk 9]
```diff
@@ -87,7 +87,89 @@ media_player:
...
+
+Set the "High-pass mode" slider of the speaker in Hz.
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| entity_id              | No       | The entity_id of the KEF speaker. |
+| hz                     | No       | Value of the slider (50 to 120 with steps of 5) |
+
+### Service `set_low_hz`
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
### Service `kef.set_low_hz`
```

File: source/_integrations/kef.markdown [hunk 10]
```diff
@@ -87,7 +87,89 @@ media_player:
...
+
+Set the "Sub out low-pass frequency" slider of the speaker in Hz.
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| entity_id              | No       | The entity_id of the KEF speaker. |
+| hz                     | No       | Value of the slider (40 to 250 with steps of 5) |
+
+### Service `set_sub_db`
```
Comment (frenck) [inline source/_integrations/kef.markdown]:
```suggestion
### Service `kef.set_sub_db`
```

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks for the PR @basnijholt!

Some small suggestions, could you please take a look? Thanks! 👍

**Comment by basnijholt** [comment]
Thanks @frenck, I've committed your changes!

**Review by frenck — CHANGES_REQUESTED** [review]
Well, that was fast @basnijholt! 👍

**Review by frenck — DISMISSED** [review]
Wrong button, wanted to approve it.

**Review by frenck — APPROVED** [review]

**Pull request merged**